react/features/face-centering/functions.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+import { getAppBaseUrl } from '../facial-recognition/functions';
+
+import { FACE_BOX_EVENT_TYPE, DETECT_FACE_BOX } from './constants';
+import logger from './logger';
+
+/**
+ * Sends the face box to all the other participants.
+ *
+ * @param {Object} conference - The current conference.
+ * @param  {Object} faceBox - Face box to be sent.
+ * @returns {void}
+ */
+export function sendFaceBoxToParticipants(
+        conference: Object,
+        faceBox: Object
+): void {
+    try {
+        conference.sendEndpointMessage('', {
+            type: FACE_BOX_EVENT_TYPE,
+            faceBox
+        });
+    } catch (err) {
+        logger.warn('Could not broadcast the face box to the other participants', err);
+    }
+}
+
+/**
+ * Sends the image data a canvas from the track in the image capture to the face centering worker.
+ *
+ * @param {Worker} worker - Face centering worker.
+ * @param {Object} imageCapture - Image capture that contains the current track.
+ * @param {number} threshold - Movement threshold as percentage for sharing face coordinates.
+ * @param {boolean} isHorizontallyFlipped - Indicates whether the image is horizontally flipped.
+ * @returns {Promise<void>}
+ */
+export async function sendDataToWorker(
+        worker: Worker,
+        imageCapture: Object,
+        threshold: number = 10,
+        isHorizontallyFlipped = true
+): Promise<void> {
+    if (imageCapture === null || imageCapture === undefined) {
+        return;
+    }
+
+    let imageBitmap;
+
+    try {
+        imageBitmap = await imageCapture.grabFrame();
+    } catch (err) {
+        logger.warn(err);
+
+        return;
+    }
+
+    const canvas = document.createElement('canvas');
+    const context = canvas.getContext('2d');
+
+    canvas.width = imageBitmap.width;
+    canvas.height = imageBitmap.height;
+    context.drawImage(imageBitmap, 0, 0);
+
+    const imageData = context.getImageData(0, 0, imageBitmap.width, imageBitmap.height);
```

Comment:
According https://html.spec.whatwg.org/multipage/imagebitmap-and-animations.html `ImageBitmap` is transferrable, so you should be able to send it directly to the worker, and then do the canvas operation there, in a non-main thread.

Reply:
don't actually need the canvas at all, since the `tf.browser.fromPixels` allows passing an ImageBitmap directly